ca/certificates.go

```diff
@@ -381,9 +387,24 @@ func NewRootCA(certBytes, keyBytes []byte, certExpiry time.Duration) (RootCA, er
 	// Calculate the digest for our Root CA bundle
 	digest := digest.FromBytes(certBytes)
 
+	// TODO(cyli): We currently do not yet support arbitrary chains of intermediates (e.g. the case of an offline root, with the CA signing with
+	// an intermediate) - we currently only only intermediates for which the first intermediate is cross-signed version of the signing (first) root
+	// for the purposes of root rotation.  If we wanted to support offline roots, rather than require at the first intermediate is the cross-signed
+	// version of the signing CA, it should *either* be a cross-signed version of the signing CA or it should be the issuer of the signing CA.
```

Comment:
There is a lot to unpack in this comment. Not sure I understand this last sentence.